src/observable/fromArrayLike.ts

```diff
@@ -0,0 +1,69 @@
+import {Scheduler} from '../Scheduler';
+import {Observable} from '../Observable';
+import {Subscriber} from '../Subscriber';
+import {Subscription} from '../Subscription';
+
+export class ArrayLikeObservable<T> extends Observable<T> {
+
+  private _mapFn: (x: number, y: any) => T;
+
+  static create<T>(arrayLike: ArrayLike<T>, mapFn: (x: number, y: any) => T, thisArg: any, scheduler?: Scheduler) {
```

Comment:
do we need support `thisArg`? in RxJS5, https://github.com/ReactiveX/RxJS/issues/878 unless there are corresponding native (i.e, `Array.map` vs `Observable.map`) function signature, does not accept `thisArg` as parameter. Curious this is applicable case.